### PR TITLE
faster backtest

### PR DESCRIPTION
**Pull request by Mila432**
replaced sqlitedict with diskcache and added bulk data download instead of using api for old data

### CODE REVIEW

**Comment by tntwist** [comment]
Hi getting the following error:
```
ModuleNotFoundError: No module named 'xmltodict'
```

My linter shows xmltodict and diskcache as missing. Did you forget to add it to the requirements?

**Comment by tntwist** [comment]
Installed the deps myself and it works like a charm 🎉

**Comment by vitorverasm** [comment]
Better to add dependencies to `requirements.txt`:
```txt
xmltodict==0.12.0
diskcache==5.2.1
```

**Review by vitorverasm — CHANGES_REQUESTED** [review]
This works well! 🥳 

Just add dependencies to `requirements.txt`:
```txt
xmltodict==0.12.0
diskcache==5.2.1
```

**Comment by idkravitz** [comment]
Hi, please check the lint errors highlighted in workflow and fix them

File: binance_trade_bot/backtest.py [hunk 1]
```diff
@@ -39,6 +73,33 @@ def increment(self, interval=1):
     def get_fee(self, origin_coin: Coin, target_coin: Coin, selling: bool):
         return 0.0075
 
+    def get_historical_klines(self, ticker_symbol='ETCUSDT', interval='1m', target_date=None, end_date=None, limit=None,
+                              frame='daily'):
+        fromdate = datetime.strptime(target_date, "%d %b %Y %H:%M:%S")  # - timedelta(days=1)
+        r = requests.get(
+            f'https://s3-ap-northeast-1.amazonaws.com/data.binance.vision?delimiter=/&prefix=data/spot/{frame}/klines/{ticker_symbol}/{interval}/',
```
Comment (idkravitz) [inline binance_trade_bot/backtest.py]: Why not using [params argument](https://docs.python-requests.org/en/master/user/quickstart/#passing-parameters-in-urls)?

**Pull request closed without merging**